modules/ui/fields/address.js

```diff
@@ -154,16 +149,21 @@ export function uiFieldAddress(field, context) {
             .enter()
             .append('input')
             .property('type', 'text')
-            .attr('placeholder', function (d) { return field.t('placeholders.' + d.id); })
+            .attr('placeholder', function (d) {
+              var countryInserter = '';
+              if (addressFormat.customPlaceholders.indexOf(d.id) !== -1) { countryInserter = '!' + countryCode; }
+              return field.t('placeholders.' + d.id + countryInserter); })
             .attr('class', function (d) { return 'addr-' + d.id; })
             .style('width', function (d) { return d.width * 100 + '%'; });
 
         // Update
-        var addrTags = [
+        // setup dropdowns for common address tags
+        if (typeof addressFormat.dropdowns !== 'undefined') { addrTags = addressFormat.dropdowns; }
```

Comment:
Hey @Natsuyasumi I'd like to get this code merged this week, so I'm taking a look through it now..

Is the reason for `addressFormat.dropdowns` so that you can turn `block_number` ("Building No.") and `housenumber` ("Lot No.") into dropdown fields?